### DISCUSSION TITLE
considering enum type as a value object

### DISCUSSION THREAD

**Discussion by mehdihadeli** [kgrzybek/modular-monolith-with-ddd#183] [Q&A] [answered]
Hi @kgrzybek,
Should we consider `enum types` in our domain model as a `enumeration class` and `value object`?
For your enumeration class you don't inherit from Enumeration class, like [jimmy bogard](https://lostechies.com/jimmybogard/2008/08/12/enumeration-classes/)  base class, what is the reason?

**Comment by kgrzybek** [comment, marked as answer]
Hi @mehdihadeli 

I don't treat `ValueObjects` as `Enumerations`. For me:
1. Enumeration is something simple, with no logic and as name implies - I can enumerate on this type.
2. ValueObject is more advanced thing, has logic, behavior, represents a business concept. Sometimes is enumerable (for example statuses of something), sometimes is not enumerable (like Money VO, how you will enumerate?)

**Reply by mehdihadeli** [reply]
@kgrzybek Thanks for your guide.
